src/coreclr/src/interop/comwrappers.cpp

```diff
@@ -584,7 +584,22 @@ HRESULT ManagedObjectWrapper::QueryInterface(
     // Find target interface
     *ppvObject = As(riid);
     if (*ppvObject == nullptr)
```

Comment:
The `ICustomQueryInterface` contract for regular objects is "first chance". `ICustomQueryInterface` gets called before the default implementation kicks in. This one is one is "last change" - it gets called after the default implementation kicks in.

We can talk about it tomorrow...

Reply:
Yep. For `ComWrappers`, that isn't really possible due to the GC issues. I also don't think it is the right semantics either. For the built-in case, we provide several default implementations that user's don't even have a say over, but in `ComWrappers` it is entirely driven by the author last seems right to me. I do agree the semantics are inverted here and... concerning given how we originally scoped/designed the `ComWrappers` feature.